src/backend/core/tests/test_models_teams.py

```diff
@@ -134,3 +134,85 @@ def test_models_teams_get_abilities_preset_role(django_assert_num_queries):
         "put": False,
         "manage_accesses": False,
     }
+
+
+# test trees
+
+def test_models_teams_create_root_team():
+    """Create a root team."""
+    team = models.Team.add_root(name="Root Team")
+    assert team.is_root()
+    assert team.name == "Root Team"
+
+
+def test_models_teams_create_child_team():
+    """Create a child team."""
+    root_team = models.Team.add_root(name="Root Team")
+    child_team = root_team.add_child(name="Child Team")
+    assert child_team.is_child_of(root_team)
+    assert child_team.name == "Child Team"
+    assert child_team.get_parent() == root_team
+
+
+def test_models_teams_create_grandchild_team():
+    """Create a grandchild team."""
+    root_team = models.Team.add_root(name="Root Team")
+    child_team = root_team.add_child(name="Child Team")
+    grandchild_team = child_team.add_child(name="Grandchild Team")
+    assert grandchild_team.is_child_of(child_team)
+    assert grandchild_team.name == "Grandchild Team"
+    assert grandchild_team.get_parent() == child_team
+
+
+def test_models_teams_move_team():
+    """Move a team to another parent."""
+    root_team = models.Team.add_root(name="Root Team")
+    child_team = root_team.add_child(name="Child Team")
+    new_root_team = models.Team.add_root(name="New Root Team")
+    child_team.move(new_root_team, pos="first-child")
+    child_team.refresh_from_db()
+    assert child_team.get_parent(update=True) == new_root_team
+
+
+def test_models_teams_delete_team():
+    """
+    Delete a parent team also deletes children.
```

Comment:
in practice this will force you to introduce soft delete soon enough! :sweat_smile:

Reply:
Probably ^^ but for now this feature is hidden from end users, so I keep it for later when we need it.